build-support/bin/generate_github_workflows.py

```diff
@@ -326,10 +331,22 @@ def generate() -> dict[Path, str]:
             },
         }
     )
+
+    test_cron_yaml = yaml.dump(
+        {
+            "name": "Daily Extended Python Testing",
+            # 11:45pm
```

Comment:
Same here re: adding a comment for why the time was chosen. I don't know if there's any reason to avoid running multiple github actions at exactly the same time, but if we're deliberately staggering them it would be good to explain that in code.